CaloReco/src/CaloHitMakerFast_module.cc

```diff
@@ -37,9 +37,8 @@ namespace mu2e {
             using Comment = fhicl::Comment;
             fhicl::Atom<art::InputTag> caloDigiCollection { Name("caloDigiCollection"), Comment("CaloDigi collection name") }; 
             fhicl::Atom<double>        digiSampling       { Name("digiSampling"),       Comment("Digitization time sampling") }; 
-            fhicl::Atom<double>        deltaTPulses       { Name("deltaTPulses"),       Comment("Maximum time difference between two signals ") }; 
-            fhicl::Atom<double>        pulseRatioMin      { Name("pulseRatioMin"),      Comment("Min value of energy ratio between pulses ") }; 
```

Comment:
mhmhmhmhmhm I much prefer the ratio w.r.t. to the residual for the energy check. This is what has been used so in the test beam analyses. The ratio scales nicely with the energy, while a crude cut on the residual does not.

Reply:
Nope, this is part of what created the mess. The ratio doesn't work at all for low-energy deposits, so you will multiple hits instead of combining them. More precisely, suppose I have two pulses that reconstruct at 2.0 and 2.5 MeV each, then they will fail the ratio test, and the code will reconstruct two hits with 2 MeV each in the same crystal at the same time -> the clustering algorithm overestimates the energy. Using the difference fixes that issue, and 5 MeV is large enough to combine high energy hits in all practical cases. I would strongly suggest using the residual for the trigger (analysis is different)

Reply:
This argument is not convincing me. with a 5 MeV tolerance, you will be combining a 2 MeV signal with a 5 MeV one, which is definitively not what we expect. If there is a problem with the noise simulation it should be addressed in a different way. If a ratio window of +/- 10% is not large enough (that is the value used in the test beam analysis) you should make it wider. In the future maybe we will make it a function/table w.r.t. to the pulse amplitudes, but crude residual sounds not optimal to me.